tasks/workflow/worker.go

```diff
@@ -592,11 +592,11 @@ func (w *worker) runUndeploy(ctx context.Context, t *taskExecution) error {
 }
 
 func (w *worker) runPurge(ctx context.Context, t *taskExecution) error {
+	// Set status to UNDEPLOYMENT_PURGING
+	deployments.SetDeploymentStatus(ctx, w.consulClient.KV(), t.targetID, deployments.UNDEPLOYMENT_PURGING)
```

Comment:
Should we use `checkAndSetTaskStatus` instead?
@stebenoist WDYT?

Reply:
@loicalbertin, you mean that the new status should be a TaskStatus and not a DeploymentStatus ?

Reply:
IMO it's a deployment status so we should use SetDeploymentStatus.

Reply:
Oups I was confused by those names... 
This is good to me.